frame/contracts/src/lib.rs

```diff
@@ -542,21 +541,16 @@ decl_module! {
 		/// You can instantiate contracts only with stored code.
 		pub fn put_code(
 			origin,
-			#[compact] gas_limit: Gas,
 			code: Vec<u8>
-		) -> Result {
-			let origin = ensure_signed(origin)?;
-
-			let (mut gas_meter, imbalance) = gas::buy_gas::<T>(&origin, gas_limit)?;
+		) -> dispatch::Result {
+			let _origin = ensure_signed(origin)?;
 
 			let schedule = <Module<T>>::current_schedule();
-			let result = wasm::save_code::<T>(code, &mut gas_meter, &schedule);
+			let result = wasm::save_code::<T>(code, &schedule);
```

Comment:
Why does `put_code` no longer require gas? Shoudn't it at least be priced higher than regular runtime call data since it's being put into storage?

Reply:
This is a good question.

TBH, I am not sure why it got gas in the first place. If you think about it, `put_code` is similar to a regular dispatchable function, although yeah with a rather pricy transaction byte fee. I will change the price down the line.